### PR TITLE
Upgrade rocksdb to 0.22

### PR DESCRIPTION
**Pull request by jasl**
Although Parity has recommended `paritydb` for a long time, `polkadot-sdk` still has not started the deprecation process for `rocksdb`.

I think we should give some love to the rocksdb. This PR upgrade to its latest version would help to reduce a few outdated dependencies. for example, bitflags 1.3.0

### CODE REVIEW

**Comment by jasl** [comment]
@bkchr Could you help to take a look at this?

File: kvdb-rocksdb/src/lib.rs [hunk 1]
```diff
@@ -905,7 +905,7 @@ rocksdb.db.get.micros P50 : 2.000000 P95 : 3.000000 P99 : 4.000000 P100 : 5.0000
 		assert!(settings.contains(" block_size: 323232"));
 
 		// LRU cache (default column)
-		assert!(settings.contains("block_cache_options:\n    capacity : 8388608"));
+		assert!(settings.contains("block_cache_options:\n    capacity : 115343360"));
```
Comment (jasl) [inline kvdb-rocksdb/src/lib.rs:908]: I haven't checked why it changed here, but this should be OK.

Comment (bkchr) [reply kvdb-rocksdb/src/lib.rs:908]: The cache size was probably changed? Can you maybe check that in the changelogs?

**Review by bkchr — APPROVED** [review]

File: kvdb-rocksdb/src/lib.rs (hunk 1 above)
Comment (jasl) [reply kvdb-rocksdb/src/lib.rs:908]: The change may relate to https://github.com/facebook/rocksdb/blob/main/HISTORY.md#public-api-changes-10

I can dig more if you think this is important

**Comment by jasl** [comment]
Humbly ping @ggwpez @ordian

Could you help to review this PR?

**Review by ordian — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Pull request merged**